.github/workflows/deploy.yml

```diff
@@ -26,6 +26,10 @@ on:
         required: false
         default: "https://staging-auth-v1.raspberrypi.org"
         type: string
+      react_app_base_url:
+        required: false
+        default: ''
```

Comment:
Instead of setting this to an empty string, set it to a "magic" word, like `AWOOGA` (or maybe just `unspecified`), which you can then test for it more explicitly (and maybe add some debugging here too!):

```bash
if [ "${{inputs.react_app_basename}}" = "AWOOGA" ] ; then
  echo "Using $deploy_dir as base_url"
else
  echo "Using ${{inputs.react_app_basename}} as base_url"
fi
```